### PR TITLE
Ensure passed JSON is an object.

### PR DESCRIPTION
**Pull request by ametaireau**
Otherwise, the colander validation on body (if any) may fail and throw a 500.

@leplatrem r?

### CODE REVIEW

File: cornice/util.py [hunk 1]
```diff
@@ -128,7 +128,13 @@ def extract_json_data(request):
     if request.body:
         try:
             body = simplejson.loads(request.body)
-            return body
+            if isinstance(body, dict):
```
Comment (leplatrem) [inline cornice/util.py]: What if the body is an array ?

Comment (Natim) [inline cornice/util.py]: Valid JSON can be string, array or dict. But what would be the purpose of sending such a body to cornice? Colander validators works on object only, isn't it?

Comment (leplatrem) [inline cornice/util.py]: No colander schemas can be sequences!

```
PUT /collection
[
  {"record": 1},
  {"record": 2},
  {"record": 3}
]

Comment (ametaireau) [inline cornice/util.py]: It seems that cornice only supports mapping in the way it's implemented currently.

Comment (leplatrem) [inline cornice/util.py]: Then maybe we could add a safety check in Service asserting that the given colander schema inherits from colander.MappingSchema ?

Comment (ametaireau) [inline cornice/util.py]: yeah, that's the direction I think we should take as well!

Comment (Natim) [inline cornice/util.py]: Actually schema does inherit from ``CorniceSchema`` I added an assertion to make sure it works.

**Pull request merged**